crates/sui-types/src/message_envelope.rs

```diff
@@ -113,6 +160,23 @@ where
 
         Ok(cert)
     }
+
+    pub fn epoch(&self) -> EpochId {
+        self.auth_signature.epoch
+    }
+
+    pub fn verify_signature(&self, committee: &Committee) -> SuiResult {
```

Comment:
does this need to be a public fn? or it should only be used by the verify(committee) below?

Reply:
In the long run I think so. But right now we have a few loose ends where we verify but without turning it into a VerifiedEnvelope.